lightning/src/chain/channelmonitor.rs

```diff
@@ -1125,6 +1125,30 @@ macro_rules! _process_events_body {
 }
 pub(super) use _process_events_body as process_events_body;
 
+pub(crate) struct WithChannelMonitor<'a, L: Deref> where L::Target: Logger {
```

Comment:
Yeah, I tend to agree that we should make this more enforceable. Ideally, we can just have those structs store a wrapped `Logger` instead of passing one and relying on the caller to do the wrapping. Though, I think that may mean introducing a lifetime parameter.

But if instead we make the types explicit on `pub(crate)` methods, then we may avoid some redundant wrapping. Let me take a look.